src/qt/fontenum.cpp

```diff
@@ -7,13 +7,29 @@
 
 // For compilers that support precompilation, includes "wx.h".
 #include "wx/wxprec.h"
-
 #include "wx/fontenum.h"
+#include "wx/qt/private/converter.h"
+
+#include <QtCore/QStringList>
+#include <QtGui/QFontDatabase>
 
-bool wxFontEnumerator::EnumerateFacenames( wxFontEncoding WXUNUSED(encoding),
-                bool WXUNUSED(fixedWidthOnly))
+bool wxFontEnumerator::EnumerateFacenames(wxFontEncoding WXUNUSED(encoding), bool fixedWidthOnly)
 {
-    return false;
+    QFontDatabase fontDatabase;
+    const QStringList allFonts = fontDatabase.families(QFontDatabase::Any);
+    for( QStringList::const_iterator i = allFonts.begin(); i != allFonts.end(); ++i )
+    {
+        const QString& fontFamily = *i;
+        if ( !fixedWidthOnly || fontDatabase.isFixedPitch(fontFamily) )
+        {
+            if ( !OnFacename(wxQtConvertString(fontFamily)) )
+            {
+                return false;
```

Comment:
I wonder if it shouldn't just `break;` here.
Other implementations are not completely consistent either, but `break` is used for most of them:
- `dfb` will break and return `true` if at least one facename was found;
- `msw` doesn't use `OnFacename()` AFAICS and will always return true;
- `osx` and `unix` will break and will always return true.

Reply:
I can't find where the return value of this function is ever used.  Does a value of false mean that no fonts have been found or that there has been some unspecified problem enumerating the fonts?

Reply:
> I can't find where the return value of this function is ever used.

If you refer to `EnumerateFacenames()`, then indeed, it's not exactly clear what its return value means. But that is not very relevant here, where we care about `OnFacename()` and how its return value should be treated, and new code should at least be consistent with other implementations.